---
title: "CNL GWAS PHS"
output: github_document
date: "`r format(Sys.time(), '%Y.%m.%d')`"
author: S.A. Martinez
---

**Genome-wide association analysis** of the Cornell Master elite nursery for the Preharvest Sprouting trait using the spike-wetting tests. 

> All files in referenced in this document can be downloaded from [CNL_GWAS_GP2020](https://github.com/shantel-martinez/CNL_GWAS_GP2020).  
> Users will need the following in the same working directory as this CNL_GWAS_PHS.Rmd file: [myGD.csv](https://github.com/shantel-martinez/CNL_GWAS_GP2020/blob/master/myGD.csv),  [myGM.csv](https://github.com/shantel-martinez/CNL_GWAS_GP2020/blob/master/myGM.csv), and [allMasterDatawithOHMI_June2017miss30.RData](https://github.com/shantel-martinez/CNL_GWAS_GP2020/blob/master/allMasterDatawithOHMI_June2017miss30.RData)  

This analysis follows the [CNL_Prep_Stats_PHS](https://github.com/shantel-martinez/CNL_GWAS_GP2020/blob/master/CNL_Prep_Stats_PHS.md) analysis in the [CNL_GWAS_GP2020](https://github.com/shantel-martinez/CNL_GWAS_GP2020) project. The environment should contain only the following dataframes:

```{r  warning=FALSE}
load("./CNL_Prep_Stats_PHS.RData")
rm(list= ls()[!(ls() %in% c('PHScGIDblup','PHSwGIDblup','PHSrGIDblup','PHSWhiteComplete','PHSCombComplete','PHSRedComplete'))])
```

If these dataframes are **not** in the current environment, open the [CNL_Prep_Stats_PHS.Rmd](https://github.com/shantel-martinez/CNL_GWAS_GP2020/blob/master/CNL_Prep_Stats_PHS.Rmd) file and run all coding chunks. *Be sure to have all the corresponding input files needed to run*  
OR
Download the [CNL_Prep_Stats_PHS.Rdata](https://github.com/shantel-martinez/CNL_GWAS_GP2020/blob/master/CNL_Prep_Stats_PHS.RData) files from github, save in the working directory this CNL_GWAS_PHS.Rmd is in. 

## PHS Genotype Files

Genotyping was conducted using genotyping-by-sequencing on the Illumina HiSeq 2000.  11,604 population-based SNP markers were identified after a 30% missing data and a 5% minor allele frequency filter was applied. [N. Santantonio et al., 2019](http://www.genetics.org/content/early/2019/01/24/genetics.118.301851) curated, imputed, and filtered the genotypic data. The `allMasterDatawithOHMI_June2017miss30.RData` file was given to me by N. Santantonio. 

Impute the genotpye dataframe `Ximp` and renamed it `myGD`: 

```{r eval = FALSE}
load("../Data Input/allMasterDatawithOHMI_June2017miss30.RData")
# myGD headers: SNP Marker1 Marker2 ... MarkerN
myGD<- Ximp + 1 
rm(pheno, snpInfo, X, Ximp, XnoHet, README)
write.csv(myGD, "./myGD.csv",row.names=TRUE)
```

Ximp is coded as: -1 is homozygous major allele, 0 is heterozygous, and 1 is homozygous minor allele.  
`Ximp + 1` changes the code to 0,1,2.   

This may not be the most efficent way to add in the taxa names into the first column, but I found that if I write `myGD` as a csv file, then read it back in as `myGD`, it is now a dataframe rather than a large matrix and includes the first column as the GIDs (or genotype names). 

```{r}
myGD <- read.csv("../Data Input/myGD.csv", head = TRUE,na.string=c(""," ","NA","na","NaN"))
rownames(myGD) = myGD[,1]
myGD[1:4,1:4]
```

### Summary of GIDs in study  

Summary of GIDs per location with genotypic data  



```{r}
library("dplyr")
PHSComb1 <- subset(PHSCombComplete, GID %in% myGD$X)
c<-PHSComb1 %>%
  group_by(Year, Loc) %>%
  summarize(Comb = length(unique(GID)))

PHSWhite1 <- subset(PHSWhiteComplete, GID %in% myGD$X)
w<- PHSWhite1 %>%
  group_by(Year, Loc) %>%
  summarize(White = length(unique(GID)))
PHSred1 <- subset(PHSRedComplete, GID %in% myGD$X)
r<- PHSred1 %>%
  group_by(Year, Loc) %>%
  summarize(Red = length(unique(GID))) 
a <- cbind(c,w,r) 

a %>%
  dplyr::select(Year, Loc, Comb, White, Red)
```


